Module/infoblox

xoanmi opened the pull request:
##### SUMMARY
This is a new ansible module to manage Infoblox IPAM & DNS devices.

##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME

/ansible/lib/ansible/modules/network/infoblox

##### ANSIBLE VERSION
```
ansible 2.4.0
```

##### ADDITIONAL INFORMATION
This module was well supported by community in my personal repo: https://github.com/xoanmi/ansible-module-infoblox

xoanmi committed:
Added infoblox module

xoanmi committed:
Delete README.md

xoanmi committed:
Delete LICENSE

xoanmi committed:
Delete example.yml

xoanmi committed:
Solve format issues

xoanmi committed:
Added __init__.py file

xoanmi committed:
Solve wildcard import issue

xoanmi committed:
Solved typo in Metadata secction

xoanmi committed:
Solved some issues related ansible pull request requirements

xoanmi committed:
Solved some issues related ansible pull request requirements

xoanmi committed:
Solved some issues related ansible pull request requirements

gundalow commented:
Please move this to `modules/net_tools`
The way the action are done doesn't seem to fit very well. Feel free to join `#ansible-devel` on Freenode IRC to discuss this.

The cutoff for new modules for `2.4` is Tuesday 29th, so it's unlikely this will make that release, instead this will be for `2.5`

Commit:
Moved module to net_tools
Added some minur bugfixins

gundalow commented:
Hi,
This module seems to be doing a lot of things all at once

```
"get_aliases", "get_cname", "get_a_record", "get_host", "get_network", "get_range", "get_next_available_ip", "get_fixedaddress",
 +              "get_ipv6network", "get_ptr_record", "get_srv_record", "get_auth_zone", "get_forward_zone", "get_delegated_zone",
 +              "add_alias", "add_cname", "add_host", "add_ipv6_host", "create_ptr_record", "get_txt_record", "get_network_container",
 +              "create_a_record", "create_srv_record", "create_auth_zone", "create_forward_zone", "create_delegated_zone", "create_txt_record",
 +              "create_network_container", "set_a_record", "set_name", "set_extattr", "update_a_record", "update_srv_record",
 +              "update_ptr_record", "update_cname_record", "update_auth_zone", "update_forward_zone", "update_txt_record",
 +              "update_network_container", "update_host_record", "delete_alias", "delete_cname", "delete_a_record", "delete_fixedaddress",
 +              "delete_host", "delete_ptr_record", "delete_srv_record", "reserve_next_available_ip"]
```

Also these don't feel like the right terms. I think this should be split into multiple modules, perhaps one for each record type, or perhaps the record type is an option

```
infoblox:
  username:
  password:
  server:
  network: 192.168.1.0/24
  state: present
  type: A
with_items:
  - test01.local
  - test01.local
```

gundalow commented:
Feel free to join `#ansible-devel` on Freenode IRC to discuss this more.

Commit:
Ensure PEP8 compliance

xoanmi committed:
Solved several issues detected on last commit

xoanmi committed:
Solved some issues
lib/ansible/modules/net_tools/infoblox.py:1481:1: E302 expected 2 blank lines, found 1
lib/ansible/modules/net_tools/infoblox.py:1504:1: E302 expected 2 blank lines, found 1
lib/ansible/modules/net_tools/infoblox.py:1525:1: E302 expected 2 blank lines, found 1
ERROR! module infoblox missing documentation (or could not parse documentation): Parsing produced an empty object.

xoanmi committed:
Solve sintaxis problem

xoanmi committed:
Solve errors

xoanmi committed:
correct some issues

gundalow commented:
Also there is https://github.com/infobloxopen/infoblox-ansible

gundalow commented:
@xoanmi I believe infoblox will be creating and upstreaming some modules, so you may want to put this on hold. I'm sure they would welcome your review when the modules get created.

gundalow commented:
@xoanmi In Ansible 2.5 (target release March) we've added various Infoblox modules http://docs.ansible.com/ansible/devel/module_docs/list_of_net_tools_modules.html#nios
If you install the `devel` branch of Ansible you'll get access to them today.

Also we have a PR in flight https://github.com/ansible/ansible/pull/35189 which adds:
* `nios` - lookup plugin to return nios objects to the playbook
* `nios_next_ip` - lookup plugin to return the next avaiable ip address

Would love your feedback on these, for example:
* Do they work for your use case
* Any feedback on useful extra functionality
* Any bugs
* Are the docs clear?

Feedback via https://github.com/ansible/ansible/issues/new

Therefore I'll close this.
